src/pybind/mgr/cephadm/module.py

```diff
@@ -455,6 +457,12 @@ def _get_cephadm_service(self, service_type: str) -> CephadmService:
         assert service_type in ServiceSpec.KNOWN_SERVICE_TYPES
         return self.cephadm_services[service_type]
 
+    def _get_cephadm_binary_path(self) -> str:
+        import hashlib
+        m = hashlib.sha256()
+        m.update(self._cephadm.encode())
+        return f'/var/lib/ceph/{self._cluster_fsid}/cephadm.{m.hexdigest()}'
```

Comment:
I remember some issue before with files or directories in /var/lib/ceph/{fsid}/ of the structure "something.something"  that were not meant to be daemons being marked as stray daemons and getting removed. Did that ever get fixed?

Reply:
thats this line https://github.com/ceph/ceph/blob/master/src/cephadm/cephadm#L4485.    should be good for this as it a file and not a dir

Reply:
@Daniel-Pivonka alright, perfect.